catalog/transformer/catalog-transformer-overlay/src/main/java/ddf/catalog/transformer/OverlayMetacardTransformer.java

```diff
@@ -71,29 +83,106 @@ private BinaryContent overlay(Metacard metacard, Map<String, Serializable> argum
         final BufferedImage image =
                 bufferedImageOptional.orElseThrow(() -> new CatalogTransformerException(
                         "Did not receive an image from the image supplier."));
-        final Builder imageBuilder = Thumbnails.of(image)
-                .scale(1)
-                .imageType(BufferedImage.TYPE_INT_ARGB)
-                .outputFormat(PNG);
-        rotate(imageBuilder, metacard.getLocation());
-        return createBinaryContent(imageBuilder);
+
+        List<Vector> boundary = parseBoundary(metacard.getLocation());
+        BufferedImage tile = createTileFromImageAndBoundary(image, boundary);
+        return createBinaryContent(tile);
     }
 
-    private void rotate(Builder imageBuilder, String wkt) throws CatalogTransformerException {
-        final double angle = GeometryUtils.getRotationAngle(wkt);
-        imageBuilder.rotate(angle);
+    private List<Vector> parseBoundary(String location) throws CatalogTransformerException {
+        final Geometry geometry = parseGeometry(location);
+        if (!canHandleGeometry(geometry)) {
+            throw new CatalogTransformerException("The metacard geometry is not a rectangle.");
```

Comment:
Can we update this to `The Image boundary is not a rectangle` or something that would be more meaningful to a user?